dotnet/SimpleSauce.Test/SauceOptionsTests.cs

```diff
@@ -1,13 +1,386 @@
+            SauceOptions.TimeZone.Should().BeEquivalentTo("San Francisco");
+            SauceOptions.TunnelIdentifier.Should().BeEquivalentTo(tags);
+            SauceOptions.VideoUploadOnPass.Should().BeFalse;
+        }
+
+        [TestMethod]
+        public void AcceptsChromeOptionsClass()
+        {
+            var options = new ChromeOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("chrome");
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsEdgeOptionsClass()
+        {
+            var options = new EdgeOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("MicrosoftEdge");
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsFirefoxOptionsClass()
+        {
+            var options = new FirefoxOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("firefox");
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsInternetExplorerOptionsClass()
+        {
+            var options = new InternetExplorerOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("internet explorer");
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void AcceptsSafariOptionsClass()
+        {
+            var options = new SafariOptions();
+            SauceOptions = new SauceOptions(options);
+
+            SauceOptions.BrowserName.Should().BeEquivalentTo("safari");
+            SauceOptions.SeleniumOptions.Should().BeEquivalentTo(options);
+        }
+
+        [TestMethod]
+        public void CreatesDefaultBuildName()
+        {
+            Environment.SetEnvironmentVariable("BUILD_TAG", "Not Empty");
+            Environment.SetEnvironmentVariable("BUILD_NAME", "TEMP BUILD");
+            Environment.SetEnvironmentVariable("BUILD_NUMBER", "11");
+            
+            SauceOptions.BuildName.Should().BeEquivalentTo("TEMP BUILD: 11");
+        }
+
+        [TestMethod]
+        public void MergesCapabilitiesFromDictionary()
+        {
+            //I don't think this is a use case with .NET bindings
+            //var capabilities = new Dictionary<string, object>();
+            //// Add all the capabilities here
+            //capabilities.Add("key", "value");
+            //SauceOptions.SetCapabilities(capabilities);
+            //SauceOptions.Build.Should().BeEquivalentTo(build);
+        }
+
+        [TestMethod]
+        public void ParsesCapabilitiesFromW3CValues()
```

Comment:
@titusfortner I'm not sure what this test is supposed to be doing? Seems like it's missing some steps?